cmake/sdp.cmake

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+function(sdp_assembly_generate hrt_srcs)
+  set(hrt_msg "Generating ASM files for Hard Real Time files.")
+  set(hrt_opts -g0 -P -fno-ident -fno-verbose-asm)
+
+  # Compiler does not know how to build for this platform so we export
+  # all the flags used in this zephyr build to the external build system.
+  zephyr_get_compile_options_for_lang(C options)
+  zephyr_get_compile_definitions_for_lang(C defines)
+  zephyr_get_include_directories_for_lang(C includes)
+  zephyr_get_system_include_directories_for_lang(C sys_includes)
+  # Replace "-I" with "-isystem" to treat all Zephyr headers as system headers
+  # that do not trigger -Werror.
+  string(REPLACE "-I" "-isystem" includes "${includes}")
+
+  set(compiler_options ${defines} ${options} ${includes} ${sys_includes})
+
+  # Define the asm_gen target that depends on all generated assembly files
+  add_custom_target(asm_gen
+    COMMENT ${hrt_msg}
+  )
+
+  foreach(hrt_src ${hrt_srcs})
+    if(IS_DIRECTORY ${hrt_src})
+      message(FATAL_ERROR "sdp_assembly_generate() was called on a directory")
+    endif()
+    get_filename_component(src_filename ${hrt_src} NAME_WE)  # filename without extension
+    add_custom_command(TARGET asm_gen
```

Comment:
this will be a problem if the `sdp_assembly_generate()` function is called multiple times, for example like this:
```
sdp_assembly_generate(foo.c)
sdp_assembly_generate(bar.c)
```

because there can only be a single `asm_gen` target created.

This might be an acceptable limitation for now, but could you please make this check and provide a use-ful error to the user, for example a check like this:
```
if(TARGET asm_gen)
    message(FATAL_ERROR "sdp_assembly_generate() already called, please note that 
                    sdp_assembly_generate() must be called only once with a list of all source files."
    )
endif()
```
Note, line length not checked in above example.

Reply:
The message with `FATAL_ERROR` has been added.